src/polycubed/src/cube.cpp

```diff
@@ -133,6 +152,56 @@ std::shared_ptr<PortIface> Cube::add_port(const std::string &name,
     throw;
   }
 
+  if (get_shadow()) {
+    // create a veth and move it in the namespace
+    std::string name_peerB = get_name() + "-" + name;
+    std::shared_ptr<Veth> veth = std::make_shared<Veth>(Veth::create(name, name_peerB));
+    VethPeer peerA = veth->get_peerA();
+    VethPeer peerB = veth->get_peerB();
+
+    peerB.set_status(IFACE_STATUS::UP);
+    peerA.set_namespace(prefix_ns + get_name());
+    peerA.set_status(IFACE_STATUS::UP);
+    if (conf.count("ip") && conf.count("netmask")) {
+      int prefix = polycube::service::utils::get_netmask_length(conf.at("netmask").get<std::string>());
+      peerA.set_ip(conf.at("ip").get<std::string>(), prefix);
+    }
+    if (conf.count("mac")) {
+      peerA.set_mac(conf.at("mac").get<std::string>());
+    }
+
+    std::string name2 = prefix_port + name;
+    nlohmann::json conf2 = nlohmann::json::object();
+    conf2["name"] = name2;
+    conf2["peer"] = name_peerB;
+
+    auto id2 = allocate_port_id();
+    std::shared_ptr<PortIface> port2;
+
+    switch (type_) {
+    case CubeType::TC:
+      port2 = std::make_shared<PortTC>(*this, name2, id2, conf2);
+      break;
+    case CubeType::XDP_SKB:
+    case CubeType::XDP_DRV:
+      port2 = std::make_shared<PortXDP>(*this, name2, id2, conf2);
+      break;
+    }
+
+    ports_by_name_.emplace(name2, port2);
```

Comment:
No it will not be displayed, because there are two maps containing the ports (which have the same name) one in cube.cpp (this) and one in libs/polycube/include/....services/cube.h.
for the show command, polycube uses the map in cube.h

Reply:
You're right, I forgot about it.